examples/imagenet/imagenet_lib.py

```diff
@@ -191,58 +159,78 @@ def _prepare(x):
   return jax.tree_map(_prepare, xs)
 
 
-def create_input_iter(batch_size, image_size, dtype, train, cache):
-  ds = input_pipeline.load_split(
-      batch_size, image_size=image_size, dtype=dtype, train=train, cache=cache)
-  it = map(prepare_tf_data, ds)
+def _create_input_iter(dataset_builder, batch_size, image_size, dtype, train,
+                       cache):
+  ds = input_pipeline.create_split(
+      dataset_builder, batch_size, image_size=image_size, dtype=dtype,
+      train=train, cache=cache)
+  it = map(_prepare_tf_data, ds)
   it = jax_utils.prefetch_to_device(it, 2)
   return it
 
 
 # flax.struct.dataclass enables instances of this class to be passed into jax
 # transformations like tree_map and pmap.
 @flax.struct.dataclass
-class TrainState:
+class _TrainState:
   step: int
   optimizer: optim.Optimizer
   model_state: nn.Collection
   dynamic_scale: optim.DynamicScale
 
 
-def restore_checkpoint(state):
-  return checkpoints.restore_checkpoint(FLAGS.model_dir, state)
+def _restore_checkpoint(model_dir, state):
+  return checkpoints.restore_checkpoint(model_dir, state)
 
 
-def save_checkpoint(state):
+def _save_checkpoint(model_dir, state):
   if jax.host_id() == 0:
     # get train state from the first replica
     state = jax.device_get(jax.tree_map(lambda x: x[0], state))
     step = int(state.step)
-    checkpoints.save_checkpoint(FLAGS.model_dir, state, step, keep=3)
+    checkpoints.save_checkpoint(model_dir, state, step, keep=3)
 
 
-def sync_batch_stats(state):
+def _sync_batch_stats(state):
   """Sync the batch statistics across replicas."""
   avg = jax.pmap(lambda x: lax.pmean(x, 'x'), 'x')
   return state.replace(model_state=avg(state.model_state))
 
 
-def main(argv):
-  if len(argv) > 1:
-    raise app.UsageError('Too many command-line arguments.')
+def train_and_evaluate(model_dir: str, batch_size: int, num_epochs: int,
+                       learning_rate: float, momentum: float, cache: bool,
+                       half_precision: bool,
+                       num_train_and_eval_steps: int = -1,
+                       disable_checkpointing: bool = False):
+  """Runs model training and evaluation loop.
+
+  Args:
+    model_dir: Directory where the checkpoints and tensorboard summaries
+      should be written to.
+    batch_size: Batch size of the input.
+    num_epochs: Number of epochs to cycle through before stopping.
+    learning_rate: Learning rate for the momentum optimizer.
```

Comment:
This isn't accurate. This argument is "the learning rate in case you have batch size 256. The effective learning rate is scaled linearly to the batch size"

Reply:
oh no... Thanks for catching (will make the necessary changes in #289 as well)